Expensify/App
# Refactor the WorkspaceSettingsPage

cc @HorusGoul @tgolen 

### Details


This PR refactors some bad smells:
1. It moves API actions from `WorkspaceSettingsPage.js` to `actions/Policy.js`
2. The actions do not return Promise anymore as this is not required in this case
3. Use Onyx as source of truth

Additionally, I have added a small change in the flow (which we can discuss if we want to keep) which is the avatar being saved in the policy when it is uploaded. Currently, user needs to upload the avatar, wait for it to load and then click `Save` to actually save the avatar into the policy. 

After this refactoring, the avatar flow is independent on the name/currency flow, so the new picture is saved to our database once it is uploaded. 

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/180546

### Tests

This is a refactoring of the Workspace settings editor. We need to make sure uploading a new avatar for workspace or changing name/currency still works as expected.

1. Go to your workspace's `General settings`.
2. Click on the avatar and pick some image to use as avatar and upload it. 
3. Success Growl message should show indicating the upload has been successful.
4. Exit the right hand side panel, reload the page (potentially even clearing your cache) and make sure the workspace has the new picture as its avatar.
5. Go back to the workspace settings and change name of the workspace. Click `Save`.
6. Exit the setting page, reload the page and make sure the name of the settings is changed.
7. Lastly, go back to the settings page, click on the Avatar and remove it. Make sure it is removed and the placeholder image is shown (even after reloading).

### QA Steps


Same as tests above.


### Tested On


### Screenshots


#### Web



https://user-images.githubusercontent.com/36083550/138750619-a624f6b7-a6f3-4b42-8016-2a5dd982c1fb.mov


#### Mobile Web


#### Desktop


#### iOS


#### Android
